common/services/request-handler.js

```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+
+let allow = true;
+let aborts = [];
+let currentid = 0;
+
+const removeController = (id) => {
+    const index = aborts.findIndex(object => {
+        return object.id === id;
+    });
+    if (!id) {
+        return;
+    }
+    aborts.splice(index);
+    return;
+}
+
+export const disableRequests = () => {
+    allow = false;
+    aborts.forEach(item => {
+        item.controller.abort();
+    })
+    return;
+}
+
+export const initializeInterceptor = (core) => {
+    core.http.intercept({
+        responseError: (httpErrorResponse, controller) => {
+            if (
+                httpErrorResponse.response?.status === 401
+            ) {
+                disableRequests();
+                setTimeout(() => window.location.reload(), 1000);
+            }
+        },
+    });
+}
+
+export const request = async (options = '') => {
+    if (!allow) {
+        return Promise.reject('Requests are disabled');
+    }
+    if (!options.method | !options.url) {
+        return Promise.reject("Missing parameters")
+    }
+    const requestId = currentid;
+    currentid++;
+    const abort = new AbortController();
+    if (allow) {
+        try {
+            aborts.push({ id: requestId, controller: abort });
+            const requestData = await axios(options);
+            removeController(requestId);
+            return Promise.resolve(requestData);
+        }
+        catch (e) {
+            return Promise.reject(e);
```

Comment:
**nitpick**: the requests done via axios are not managing when there is a request failing due to the session expired.

The security plugin, implement a request interceptor in the `core.http` service that reloads the page when a request with status code 401 is detected.

Reply:
The service is intended to manage the requests and prevent doing more of them with the interceptor configured before. 

The problem using `core.http` is that it will reload the page with every 401 code received. That include some codes that we receive from the Wazuh API, leading to a healthcheck loop. 

This approach intends to allow keep making requests if the 401 is received from the Wazuh API, but disable them if there's a 401 received on Kibana/Opensearch Dashbords (i.e. due to session expired), because those uses `core.http`